pkg/core/native/native_neo.go

```diff
@@ -839,19 +839,10 @@ func (n *NEO) CalculateNEOHolderReward(d *dao.Simple, value *big.Int, start, end
 }
 
 func (n *NEO) registerCandidate(ic *interop.Context, args []stackitem.Item) stackitem.Item {
-	pub := toPublicKey(args[0])
-	if !ic.IsHardforkEnabled(config.HFEchidna) {
-		ok, err := runtime.CheckKeyedWitness(ic, pub)
-		if err != nil {
-			panic(err)
-		} else if !ok {
-			return stackitem.NewBool(false)
-		}
-	}
 	if !ic.VM.AddGas(n.getRegisterPriceInternal(ic.DAO)) {
 		panic("insufficient gas")
 	}
-	var err = n.RegisterCandidateInternal(ic, pub)
+	var err = n.RegisterCandidateInternal(ic, toPublicKey(args[0]))
```

Comment:
This one is a special case, let's not rewrite the git history for this line and keep `pub := toPublicKey(args[0])` declaration (although `pub` is not used anywhere).